File: models/invite.go
package models

// Invite represents a server invite in API responses.
type Invite struct {
	ID            string  `json:"id"`
	Code          string  `json:"code"`
	ChannelID     string  `json:"channel_id"`
	CreatorID     string  `json:"creator_id"`
	MaxUses       *int    `json:"max_uses"`
	UseCount      int     `json:"use_count"`
	MaxAgeSeconds *int    `json:"max_age_seconds"`
	ExpiresAt     *string `json:"expires_at,omitempty"`
	CreatedAt     string  `json:"created_at"`
}

// CreateInviteRequest is the body for POST /api/v1/server/invites.
type CreateInviteRequest struct {
	ChannelID     string `json:"channel_id"`
	MaxUses       *int   `json:"max_uses"`
	MaxAgeSeconds *int   `json:"max_age_seconds"`
}

// AcceptOnboardingRequest is the body for POST /api/v1/onboarding/accept.
type AcceptOnboardingRequest struct {
	AcceptedRules bool `json:"accepted_rules"`
}

File: models/invite.ts
/** Server invite returned by the API. */
export interface Invite {
  id: string;
  code: string;
  channel_id: string;
  creator_id: string;
  max_uses: number | null;
  use_count: number;
  max_age_seconds: number | null;
  expires_at: string | null;
  created_at: string;
}

/** Body for POST /api/v1/server/invites. */
export interface CreateInviteRequest {
  channel_id: string;
  max_uses?: number | null;
  max_age_seconds?: number | null;
}

/** Body for POST /api/v1/onboarding/accept. */
export interface AcceptOnboardingRequest {
  accepted_rules: boolean;
}
